### PR TITLE
chore(testrunner): introduce tests for TestRunner

### PR DESCRIPTION
**Pull request by aslushnikov**
This adds some basic tests for the test runner.

### CODE REVIEW

File: utils/testrunner/TestRunner.js [hunk 1]
```diff
@@ -64,7 +64,7 @@ class UserCallback {
         const from = frame.indexOf('(');
         frame = frame.substring(from + 1, frame.length - 1);
       } else {
-        frame = frame.substring('at '.length + 1);
+        frame = frame.substring('at '.length);
```
Comment (aslushnikov) [inline utils/testrunner/TestRunner.js:67]: this was an actual bug

File: utils/testrunner/TestRunner.js [hunk 2]
```diff
@@ -348,14 +348,21 @@ class TestRunner extends EventEmitter {
...
+    this._runningPass = new TestPass(this, this._rootSuite, runnableTests, this._parallel, this._breakOnFailure);
+    const termination = await this._runningPass.run();
+    this._runningPass = null;
     if (termination)
       this.emit(TestRunner.Events.Terminated, termination.message, termination.error);
     else
       this.emit(TestRunner.Events.Finished);
   }
 
+  terminate() {
```
Comment (aslushnikov) [inline utils/testrunner/TestRunner.js:360]: I'm adding this here as a testRunner API. I'll probably move process signal handling to the Reporter that will use this api.

File: package.json [hunk 3]
```diff
@@ -15,7 +15,7 @@
     "funit": "BROWSER=firefox node test/test.js",
     "debug-unit": "node --inspect-brk test/test.js",
     "test-doclint": "node utils/doclint/check_public_api/test/test.js && node utils/doclint/preprocessor/test.js",
-    "test": "npm run lint --silent && npm run coverage && npm run test-doclint && npm run test-node6-transformer && npm run test-types",
+    "test": "npm run lint --silent && npm run coverage && npm run test-doclint && npm run test-node6-transformer && npm run test-types && node utils/testrunner/test/test.js",
```
Comment (JoelEinbinder) [inline package.json:18]: I like that we test the test runner with the test runner to make sure that we tested the test runner correctly.

File: utils/testrunner/test/testrunner.spec.js [hunk 4]
```diff
@@ -0,0 +1,522 @@
...
+    });
+  });
+
+  describe('TestRunner parallel', () => {
+    it('should run tests in parallel', async() => {
+      const log = [];
+      const t = new TestRunner({parallel: 2});
+      t.it('uno', async state => {
+        log.push(`Worker #${state.parallelIndex} Starting: UNO`);
+        await Promise.resolve();
```
Comment (JoelEinbinder) [inline utils/testrunner/test/testrunner.spec.js:354]: up to you, but relying on the number of ticks that await Promise.resolve() takes seems like a good way to get burned.

**Review by JoelEinbinder — APPROVED** [review]

File: utils/testrunner/test/testrunner.spec.js (hunk 4 above)
Comment (aslushnikov) [reply utils/testrunner/test/testrunner.spec.js:354]: Yeah I was hesitant on this one as well. Let's wait to get burned and iterate from there

File: package.json (hunk 3 above)
Comment (aslushnikov) [reply package.json:18]: Like a big boys!

**Pull request merged**